pkg/util/nsenter/nsenter.go

```diff
@@ -119,8 +130,19 @@ func (ne *Nsenter) AbsHostPath(command string) string {
 	return path
 }
 
-// SupportsSystemd checks whether command systemd-run exists
+// SupportsSystemd checks whether optional command systemd-run exists
 func (ne *Nsenter) SupportsSystemd() (string, bool) {
-	systemdRunPath, ok := ne.paths["systemd-run"]
-	return systemdRunPath, ok && systemdRunPath != ""
+	systemdBinary := "systemd-run"
+	// search for "systemd-run" command if not performed
+	// systemd-run is optional, bailout if we don't find any of the other binaries
+	if ne.hasSystemd == nil {
+		found := ne.findBinary(systemdBinary)
+		ne.hasSystemd = &found
+	}
+
+	var systemdRunPath string
+	if *ne.hasSystemd {
+		systemdRunPath = ne.paths[systemdBinary]
+	}
+	return systemdRunPath, len(systemdRunPath) != 0
```

Comment:
we have a hasSystemd and and we are checking string length, is there every a case where findBinary returns true but ne.path[systemdBinary] is empty?